src/cider_nrepl/main.clj

```diff
@@ -28,17 +28,36 @@
   [middleware-var-strs]
   (into [] mw-xf middleware-var-strs))
 
+(defn- build-handler
+  [{:keys [handler middleware]}]
+  (let [handler (or handler nrepl.server/default-handler)]
+    (if middleware
+      (apply handler (->mw-list middleware))
+      (handler))))
+
 (defn start-nrepl
-  [opts]
-  (let [{:keys [handler middleware bind port] :or {port 0 bind "::"}} opts
+  "Starts a socket-based nREPL server. Accepts a map with the following keys:
+ 
+   * :port — defaults to 0, which autoselects an open port
+
+   * :bind — bind address, by default \"::\" (falling back to \"localhost\" if
+     \"::\" isn't resolved by the underlying network stack)
+
+   * :handler — the nREPL message handler to use for each incoming connection;
+     defaults to the result of `(nrepl.server/default-handler)`
 
-        handler (cond-> (or handler nrepl.server/default-handler)
-                  middleware (apply (->mw-list middleware)))
+   * :middleware - a sequence of vars or string which can be resolved to vars,
+     representing middleware you wish to mix in to the nREPL handler. Vars can
+     resolve to a sequence of vars, in which case they'll be flattened into the
+     list of middleware."
+  [{:keys [bind port] :as opts}]
+  (let [handler
+        (build-handler opts)
 
         {:keys [server-socket port] :as server}
         (nrepl.server/start-server :handler handler
-                                   :bind bind
-                                   :port port)
+                                   :bind (or bind "localhost")
```

Comment:
Doesn't bind work with `nil`?

Reply:
So, I think what will happen here is an exception when this goes through travis. From `clojure.tools.nrepl.server`:

```
 (let [bind-addr (if bind
                    (InetSocketAddress. ^String bind ^Integer port)
                    (let [local (InetSocketAddress. "::" port)]
                      (if (.isUnresolved local)
                        (InetSocketAddress. "localhost" port)
                        local)))
```

`nil` causes `(InetSocketAddress. "::" port)` to be invoked, which throws an exception in travis - I presume due to IPv6 not being enabled. If we default to localhost, we dodge that particular idiosyncrasy.

Reply:
Sorry I should clarify: I made this change after experiencing the exception in travis. Not a thought experiment, but an observation :)

Reply:
I see. I was just wondering if this won't message up something for people who have disabled IPv4. Probably not a big deal now, I'll fixed this once we finally move the project to https://github.com/nrepl/nREPL

Reply:
(tools.nrepl is practically dead, so we've migrated the project out of clojure contrib to revive it)